### PR TITLE
Revert "MAINT: Implement API changes for randomgen-derived code"

### PR DESCRIPTION
**Pull request by bashtage**
This reverts commit 17e0070df93f4262908f884dca4b08cb7d0bba7f to readd the PCG generators following the email confirmation of the license change from @imneme sent to @bashtage and @rkern.

### CODE REVIEW

**Comment by bashtage** [comment]
Was a bear to pull out of a block of squashed commits.

**Comment by bashtage** [comment]
Is the confusion about reading PCG? or the challenge of pulling one PR out of a bundle of 20 in 17e0070df93f4262908f884dca4b08cb7d0bba7f ?

If it is the former, I received an email today from Prof. O'Neill agreeing to the license change. 

If it is the latter, take a look at the conflicts and moves in 

```
git revert 17e0070df93f4262908f884dca4b08cb7d0bba7f 
```

run against the tip of this branch.

File: doc/source/reference/random/performance.py [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-         'Normals': 'standard_normal(size=1000000)',
-         'Exponentials': 'standard_exponential(size=1000000)',
-         'Gammas': 'standard_gamma(3.0,size=1000000)',
-         'Binomials': 'binomial(9, .1, size=1000000)',
-         'Laplaces': 'laplace(size=1000000)',
-         'Poissons': 'poisson(3.0, size=1000000)', }
+funcs = OrderedDict()
+integers = 'integers(0, 2**{bits},size=1000000, dtype="uint{bits}")'
+funcs[ '32-bit Unsigned Ints'] = integers.format(bits=32)
+funcs[ '64-bit Unsigned Ints'] = integers.format(bits=64)
```
Comment (rkern) [inline doc/source/reference/random/performance.py]:
```suggestion
funcs['64-bit Unsigned Ints'] = integers.format(bits=64)
```

File: doc/source/reference/random/performance.py [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
-         'Uniforms': 'random(size=1000000)',
-         'Normals': 'standard_normal(size=1000000)',
-         'Exponentials': 'standard_exponential(size=1000000)',
-         'Gammas': 'standard_gamma(3.0,size=1000000)',
-         'Binomials': 'binomial(9, .1, size=1000000)',
-         'Laplaces': 'laplace(size=1000000)',
-         'Poissons': 'poisson(3.0, size=1000000)', }
+funcs = OrderedDict()
+integers = 'integers(0, 2**{bits},size=1000000, dtype="uint{bits}")'
+funcs[ '32-bit Unsigned Ints'] = integers.format(bits=32)
```
Comment (rkern) [inline doc/source/reference/random/performance.py]:
```suggestion
funcs['32-bit Unsigned Ints'] = integers.format(bits=32)
```

**Review by rkern — COMMENTED** [review]
Tiny typos.

File: doc/source/reference/random/performance.py (hunk 2 above)
Comment (bashtage) [reply doc/source/reference/random/performance.py]: Thanks, fixed.

File: doc/source/reference/random/performance.py (hunk 1 above)
Comment (bashtage) [reply doc/source/reference/random/performance.py]: Thanks, fixed.

**Comment by mattip** [comment]
Sorry it was so hard

**Comment by bashtage** [comment]
I now understand why many people don't like a squash rebase.

**Pull request merged**